ivy/functional/ivy/experimental/set.py

```diff
@@ -0,0 +1,57 @@
+# global
+from typing import Union, Optional
+
+# local
+import ivy
+from ivy.utils.backend import current_backend
+from ivy.func_wrapper import (
+    to_native_arrays_and_back,
+    handle_out_argument,
+    handle_nestable,
+    handle_array_like_without_promotion,
+)
+from ivy.utils.exceptions import handle_exceptions
+
+
+@to_native_arrays_and_back
+@handle_out_argument
+@handle_array_like_without_promotion
+@handle_nestable
+@handle_exceptions
+def difference(
```

Comment:
I think we can make a superset for this function as it is handled differently on the Tensorflow side, we can add `assume_unique` argument. And handle the case of multi-dimensional arrays as `TensorFlow supports this.
```python
import tensorflow as tf

set1 = tf.sparse.SparseTensor(indices=[[0, 0], [1, 2], [2, 3]], values=[1, 2, 3], dense_shape=[3, 4])
set2 = tf.sparse.SparseTensor(indices=[[0, 0], [2, 2], [2, 3]], values=[1, 2, 3], dense_shape=[3, 4])

diff = tf.sets.difference(set1, set2)
print(diff)
```